website/redirects.next.js

```diff
@@ -1272,4 +1272,9 @@ module.exports = [
     destination: '/docs/k8s/operations/tls-on-existing-cluster',
     permanent: true,
   },
+  {
+    source: '/docs/agent/options',
+    destination: '/docs/agent/config',
+    permanent: true,
+  },
```

Comment:
```suggestion
  {
    source: '/docs/agent/options',
    destination: '/docs/agent/config',
    permanent: true,
  },
```
This is going to break **a lot** of external links outside of Consul.io (Learn, random links scattered across the Internet, etc) that point to config options that used to be on this page. 😞

Reply:
I'm not seeing the diff in that suggestion. Is it a whitespace change?

I think you raise an excellent point. We should strive to make sure those links continue to work.  I think it should be a fair assumption that most links point at refs that were moved to the `config-files` page. There should be fewer links to refs on the `cli-flags` page because there are fewer command line flags, and before most of the time we suggest config files instead.  The inline links edited in this PR seem to confirm that assumption, there are a lot more that point at the new `config-files` page.

I think we can solve this problem by changing the redirect to `destination: /docs/agent/config/config-files`. That should preserve many of the links. Is that what you were suggesting above?


I tested this out with one of the existing redirects. If you click this link, https://www.consul.io/docs/acl/acl-system#acl-policies you should see the browser redirect, and still take you to the ACL Policies section. It worked for me in Chrome, and I hope all other browsers work the same way.

Do you think that is sufficient to preserve most links? A few that link to CLI flags will be broken, but I personally think that's a reasonable cost for the value we get from this change.

Reply:
> I'm not seeing the diff in that suggestion. Is it a whitespace change?
Sorry, I was hoping to suggest an improvement to the redirect that matched on the fragment portion of the URL, and redirected the user to the appropriate page based on whether a `-` was present in the fragment (e.g., `#_raft_trailing_logs`).

I realized that the fragment is not actually part of the URL that is sent to the server, so that isn't even a possible solution.

> I think we can solve this problem by changing the redirect to destination: /docs/agent/config/config-files. That should preserve many of the links. Is that what you were suggesting above?

I think that's a better solution over redirecting to `/config`.

I wonder if its possible to add extra JavaScript to the `/config-files` page that looks at `window.location` and redirects to the user to `/cli-flags` if the fragment/anchor starts with `#_`. 🤔

Reply:
I agree with @dnephin , I think it's worth to take a hit for the increased value we get out of this change.

Reply:
>Do you think that is sufficient to preserve most links? A few that link to CLI flags will be broken, but I personally think that's a reasonable cost for the value we get from this change.

I also agree with @dnephin, assuming we're sure this is the new page structure we want to build upon long term (where CLI flags and config file parameters are on different pages).

One alternative I thought of which I think is worse, but wanted to raise nonetheless because it would optimize for not breaking anything (at the expense of pointing people to outdated docs) ... We could take advantage of the fact that we have versioned docs and redirect `/docs/agent/options` to an older version of the page (e.g., `/docs/v1.10.x/agent/options`). Newer links wouldn't get redirected (because they would never point to the now non-existent `/docs/agent/options` page). Older links would go to the older page and not be broken (presumably - I haven't tested this).

Reply:
@jkirschner-hashicorp I worry that behavior will generate confusion 😟

Reply:
Good ideas! The javascript redirect could work. We could also put a banner at the top of the new `config-files` page to say something like "looking for command line flags? Go here". And I think we could make the `here` link include the fragment, so they'd go immediately to the right section by clicking that link. That might be slightly easier than automatically redirecting them. I think either is a good option for mitigating the cost. 

> We could take advantage of the fact that we have versioned docs and redirect /docs/agent/options to an older version of the page 

That seems like it would also work. I don't have a strong preference either way, lots of good options.

Reply:
@dnephin that might work with the `<Callouts>` [component](https://react-components-hashicorp.vercel.app/components/callouts)

Reply:
I actually like @jkirschner-hashicorp suggestion about versioning. I've made "breaking changes" in the docs between versions in the past and I most people seem to have found it acceptable. But I'm on board with whatever everyone else decides.